faust-macro-lib/src/lib.rs

```diff
@@ -0,0 +1,225 @@
+#![warn(
+    clippy::all,
+    // clippy::restriction,
+    clippy::pedantic,
+    clippy::nursery,
+    // clippy::cargo
+)]
+#![allow(clippy::missing_panics_doc)]
+
+use faust_build::{
+    faust_arg::{FaustArg, FaustArgsToCommandArgs},
+    faust_utils::json_path_from_dsp_path,
+};
+use faust_json::deserialize::FaustJson;
+use quote::{format_ident, quote};
+use std::{
+    fs::{self, File},
+    io::{BufReader, BufWriter, Write},
+    path::Path,
+    process::Command,
+};
+use syn::{parse_str, Ident};
+use tempfile::NamedTempFile;
+
+fn strip_quotes(name: &proc_macro2::TokenTree) -> String {
+    name.to_string()
+        .strip_prefix('\"')
+        .expect("prefix is not \"")
+        .strip_suffix('\"')
+        .expect("postfix is not \"")
+        .to_string()
+}
+
+fn get_declared_value(key: &str, ts: proc_macro2::TokenStream) -> Option<String> {
+    // find the token that declares a key in the dsp file
+    let mut ii = ts.into_iter();
+    while let Some(n) = ii.next() {
+        if n.to_string() == "declare" {
+            if let Some(n) = ii.next() {
+                if n.to_string() == key {
+                    if let Some(value) = ii.next() {
+                        if let Some(semicolon) = ii.next() {
+                            if semicolon.to_string() == ";" {
+                                return Some(strip_quotes(&value));
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+    None
+}
+
+#[must_use]
+pub fn get_name_token(ts: proc_macro2::TokenStream) -> String {
+    get_declared_value("name", ts)
+        .expect("name declaration is not found.\n Expect 'declare name NAMESTRING;' in faust code.")
+}
+
+pub fn get_flags_token(ts: proc_macro2::TokenStream) -> Vec<String> {
+    get_declared_value("flags", ts).map_or_else(std::vec::Vec::new, |s| {
+        s.split_whitespace()
+            .map(std::borrow::ToOwned::to_owned)
+            .collect()
+    })
+}
+
+#[must_use]
+pub fn write_temp_dsp_file(faust_code: &str) -> NamedTempFile {
+    let temp_dsp = NamedTempFile::new().expect("failed creating temp dsp file");
+    let mut f = BufWriter::new(temp_dsp);
+    f.write_all(faust_code.as_bytes())
+        .expect("Unable to write to temp dsp file");
+    f.into_inner().expect("temp dsp error on flush")
+}
+
+pub fn faust_command(
+    temp_dsp_path: &Path,
+    struct_name: &impl ToString,
+    flags: &[impl ToString],
+) -> Command {
+    let mut args: Vec<FaustArg> = vec![
+        FaustArg::default_lang(),
+        FaustArg::default_timeout(),
+        FaustArg::DebugWarnings,
+        FaustArg::StructName(struct_name.to_string()),
+        FaustArg::Json(),
+        FaustArg::DspPath(temp_dsp_path.to_path_buf()),
+    ];
+    for arg in flags {
+        args.push(FaustArg::Custom(arg.to_string()));
+    }
+
+    let mut faust = Command::new("faust");
+    faust.args(args.to_command_args());
+    faust
```

Comment:
Can we use FaustBuild here instead of building the `std::process::Command` manually?